### ISSUE TITLE
Implement VARCHAR data type

### ISSUE THREAD

**Issue by kyukhin** [tarantool/tarantool#4088]
Subj.

Should be possible to use CHECK machinery to check field length
before during DML. Main issue - we need to implement CHECKs
in such a way to avoid tuple decode.

**Comment by kyukhin** [comment]
Blocked by #3691

**Comment by Korablev77** [comment]
TBO I don't consider the idea of validating string length using check constrains to be sensible. 'check' machinery is quite heavyweight: we should prepare VDBE, execute bytecode, finalyze it. Meanwhile, procedure of verifying string length is simple and straightforward.

**Comment by Korablev77** [comment]
Duplicate of https://github.com/tarantool/tarantool/issues/3665

**Comment by pgulutzan** [comment]
@Korablev77: It seems odd to me that you decided #4088 "Implement VARCHAR data type" is a duplicate of issue #3665 "sql: introduce strings with fixed-length type". There is one feature that they have in common -- checking for maximum length -- and I suppose that's hard because there has to be compatibility with Lua 'string', which is VARCHAR(max). However, at least a slight step toward standard compatibility would happen if the synonyms were accepted, i.e. CHARACTER VARYING and CHAR VARYING.
